src/main/java/com/theironyard/war/Game.java

```diff
@@ -0,0 +1,130 @@
+     * the pile. If a player runs out of cards at anytime, it is gameover.
+     */
+    public void runWar() {
+        ArrayList<Card> warPile = new ArrayList<Card>();
+        System.out.println("--------- Its time for WAR!!!!! ---------");
+        for (int i = 0; i < 4; i++) {
+            warPile.add(player1.giveCard());
+            warPile.add(player2.giveCard());
+            if (player1.hand.size() == 0) {
+                System.out.format("%s you are the winner!\n", player2.getName());
+                return;
+            } else if (player2.hand.size() == 0) {
+                System.out.format("%s you are the winner!\n", player1.getName());
+                return;
+            }
+        }
+        Card player1Card = player1.giveCard();
+        Card player2Card = player2.giveCard();
+        System.out.format("%s plays %s\n", player1.getName(), player1Card.toString());
+        System.out.format("%s plays %s\n", player2.getName(), player2Card.toString());
+        Rank player1Rank = player1Card.getRank();
+        Rank player2Rank = player2Card.getRank();
+        if (player1Rank.getValue() > player2Rank.getValue()) {
+            System.out.format("%s won the War!\n", player1.getName());
+            player1.hand.add(player2Card);
+            player1.hand.add(player1Card);
+            player1.hand.addAll(warPile);
+            player1.shuffleHand();
+        } else if (player1Rank.getValue() < player2Rank.getValue()) {
+            System.out.format("%s won the War!\n", player2.getName());
+            player2.hand.add(player1Card);
+            player1.hand.add(player2Card);
+            player2.hand.addAll(warPile);
+            player2.shuffleHand();
+        } else if (player1Rank.getValue() == player2Rank.getValue()) {
+            runWar();
+        } else {
+            runGame();
+        }
+    }
+
+    /**
+     * During the battle method each player draws the top card of their hand and plays it. After comparing the rank
+     * of the cards, whomever has the higher rank wins both cards and the method continues to play until someone runs
+     * out of cards to play. If the rank is the same, another method is called, runWar, to see who is the winner of
+     * that round.
+     */
+    public void battle() {
+        if (player1.hand.size() == 0 || player2.hand.size() == 0) {
+            runGame();
+        } else {
+            Card player1Card = player1.giveCard();
+            Card player2Card = player2.giveCard();
+            System.out.format("%s plays %s\n", player1.getName(), player1Card.toString());
+            System.out.format("%s plays %s\n", player2.getName(), player2Card.toString());
+            Rank player1Rank = player1Card.getRank();
+            Rank player2Rank = player2Card.getRank();
+            if (player1Rank.getValue() > player2Rank.getValue()) {
+                System.out.format("%s won the Round!\n", player1.getName());
+                player1.hand.add(player2Card);
```

Comment:
This code is almost identical to the code below and the code in war.  I would recommend adding them to an arraylist or something then just add all the cards from the arraylist to the player.